components/scg/src/scg-render.js

```diff
@@ -397,44 +398,47 @@ SCg.Render.prototype = {
         });
 
         this.d3_links.each(function (d) {
-
             if (!d.need_observer_sync && d.contentLoaded) return; // do nothing
 
             if (!d.contentLoaded) {
-                var links = {};
-                links[d.containerId] = d.sc_addr;
+                let links = {};
+                links[d.containerId] = {addr: d.sc_addr, content: d.content, contentType: d.contentType};
                 self.sandbox.createViewersForScLinks(links);
 
-                d.contentLoaded = true;
+                if (d.content.length) d.contentLoaded = true;
             }
-            else
-                d.need_observer_sync = false;
+            else d.need_observer_sync = false;
 
-            var linkDiv = $(document.getElementById("link_" + self.containerId + "_" + d.id));
-            if (!d.sc_addr) {
-                linkDiv.find('.impl').html(d.content);
-                linkDiv.find('img').css({ 'width': '100%', 'height': '100%' })
-            } else {
-                if (d.content != "") {
-                    linkDiv.find('.impl').html(d.content);
-                    linkDiv.find('img').css({ 'width': '100%', 'height': '100%' });
-                } else {
-                    d.content = linkDiv.find('.impl').html();
-                    if (d.content != "") {
-                        d.setAutoType();
-                    }
-                }
+            let linkDiv = $(document.getElementById("link_" + self.containerId + "_" + d.id));
+            if (!d.content.length) {
+                d.content = linkDiv.find('.impl').html();
+            }
+            if (!linkDiv.find('.impl').html().length) {
+                linkDiv.find('.impl').html(d.content)
             }
 
-            var g = d3.select(this).attr("style", 'opacity: ' + d.opacityElem + '; stroke: ' + d.strokeElem + '')
-
+            const imageDiv = linkDiv.find('img');
+            const pdfDiv = linkDiv.children().find('canvas');
```

Comment:
Synchronize button is stucked after creating the second link
![image](https://github.com/ostis-ai/sc-web/assets/36566977/a6373973-a9aa-4103-b500-b69e8442b307)